docs/data-flow.md

```diff
@@ -0,0 +1,103 @@
+
+The format of the request is the following:
+
+```javascript
+{
+  "user-service": PartialSignupRequest,
+  "erp-service": ErpProducerRequest,
+}
+```
+
+Neither key is required but at least one should be defined.
+
+See:
+- [ProducerRequest](https://github.com/comicrelief/data-models/blob/master/src/schemas/supporter-events/models/requests/ProducerRequest.ts)
+- [Complete examples (mocks)](https://github.com/comicrelief/data-models/blob/master/src/schemas/supporter-events/mocks/requests/ProducerRequest.mock.ts)
+- [PartialSignupRequest](https://github.com/comicrelief/data-models/blob/master/src/schemas/user-service/models/requests/SignupRequest.ts)
+- [ErpProducerRequest](https://github.com/comicrelief/data-models/blob/master/src/schemas/erp/models/requests/ProducerRequest.ts)
+
+### `user-service` key
+
+`serverless-user-service` payloads are meant to be completed by `supporter-event-service`. In particular, `supporter-event-service` will loop through `identities` and hash `identities.content` and add it as `identities.uuid` key if it is not already defined, effectively transforming a `PartialSignupRequest` (accepted by `supporter-event-service`) into a `SignupRequest` (accepted by `serverless-user-service`).
+
+See:
+- [Identity](https://github.com/comicrelief/data-models/blob/master/src/schemas/user-service/models/Identity.ts)
+- [PartialSignupRequest and SignupRequest](https://github.com/comicrelief/data-models/blob/master/src/schemas/user-service/models/requests/SignupRequest.ts)
+
+### `erp-service` key
+
+`serverless-erp-service` payloads are meant to be completed by `supporter-event-service`. You can see [check out the specific implementations on `data-model`](https://github.com/comicrelief/data-models/tree/master/src/schemas/erp/models/actions). They consists of simple object contains an array of Erp Actions.
+
+```javascript
+{
+  "actions": ErpAction[]
+}
+```
+
+See:
+- [ErpProducerRequest](https://github.com/comicrelief/data-models/blob/master/src/schemas/erp/models/requests/ProducerRequest.ts)
+- [ErpProducerRequest examples (mocks)](https://github.com/comicrelief/data-models/blob/master/src/schemas/supporter-events/mocks/requests/ProducerRequest.mock.ts)
+
+Each ErpAction should validate against the following:
+
+```javascript
+{
+  "type": ErpPayloadTypes,
+  "data": {
+    // Upstream service specific data
+    specificData1: any,
+    specificData2: any,
+
+    // The following key is likely to be required
+    // for supporter-specific payloads
+    // such as donations, prize tickets, etc.
+    // See the `Supporter UUID` section for details
+    supporterUuid: String,
+
+    // If your service doesn't have access
+    // to supporter specific information
+    // the payload will need to specify
+    // the `identityHash` key so that `supporter-event-service`
+    // will be able to complete the payload
+    identityHash: String
```

Comment:
where is the logic for that key in `supporter-event-service`? or is it something that is still in progress?

Reply:
No logic in place for this now - this PR acts like a proposal, but it will need to be implemented because `serverless-prize-platform` needs it.